standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/DatabaseProduct.java

```diff
@@ -776,4 +782,21 @@ public Configuration getConf() {
   public void setConf(Configuration c) {
     myConf = c;
   }
+
+  /**
+   * lockInternal() and {@link #unlockInternal()} are used to serialize those operations that require
+   * Select ... For Update to sequence operations properly.  In practice that means when running
+   * with Derby database.  See more notes at class level.
+   */
+  public void lockInternal() {
+    if(isDERBY()) {
```

Comment:
nit: space

note: SqlRetryHandler has duplicated these methods. Could we please kill the dups